File: Numbers/java/PI_to_N.java
//IMPORT REQUIRED FILES
import java.util.Scanner;
import java.math.BigDecimal;


public class Main
{
	public static void main(String[] args) 
	{
    //CREATE SCANNER OBJECT
		Scanner scanner = new Scanner(System.in);
		System.out.println("Enter number of digits of PI you would like to see: ");
		int digits = scanner.nextInt();
		float pi  = 0;
		float denom = 1;
		float multiplier = 4;
    
    //CALCULATE PI
		for(long i = 0; i < 10000000; i++)
		{
			if (i%2!=0)
			{
				pi+=1/denom;
				denom+=2;
			}
			
			else
			{
				pi-=1/denom;
				denom+=2;
			}
		}
		pi*= -multiplier;
    
    //PRINT PI TO DESIGNATED PRECISION
		System.out.println(BigDecimal.valueOf(pi).setScale(digits, BigDecimal.ROUND_HALF_DOWN).floatValue());

	}
}

File: Numbers/cpp/PI_to_N.cpp
//INLCUDE REQUIRED HEADER FILES
#include <iostream>
#include <iomanip>

using namespace std;

int main()
{
  //SET VALUES TO BE USED IN CALCULATIONS
	float denom = 1;
	float multiplier = 4;
	double pi = 0;
  //CALCULATE PI
	for (long i = 0; i < 10000000; i++)
	{
		if (i%2 != 0)
		{
			pi+=1/denom;
			denom+=2;
		}
		else
		{
			pi-=1/denom;
			denom+=2;
		}
	}
	pi*=multiplier;
  
	int digits = 0;
  //GET USER INPUT FOR NUMBER OF DIGITS 
	cout<<"How many digits of PI would you like to see? "<<endl;
	cin>>digits;
  //SET IOSTREAM PRECISION TO DESIGNATED NUMBER
	cout<<setprecision(digits+1);
  //OUTPUT
	cout<<-pi<<endl;

	return 0;
}
